src/coffee-script.coffee

```diff
@@ -62,7 +62,8 @@ exports.compile = compile = withPrettyErrors (code, options) ->
   for fragment in fragments
     # Update the sourcemap with data from each fragment
     if options.sourceMap
-      if fragment.locationData
+      # Do not include empty, whitespace, or semicolon-only fragments.
+      if fragment.locationData and not /^[;\s]*$/.test fragment.code
```

Comment:
I have looked at a few constructs (not exhaustive by any means), and I have not found source mapping problems with those punctuation.  In other words, when they're mapped, they seem to be mapped between appropriate locations in the .coffee and .js (for example the ':' in a generated tenerary operator maps to an inline "if" in coffeescript).

Have you seen problems mapping fragments with those punctuation?  I'm inclined to just add cases where we've seen a problematic example.